samples/algorithms/MajoranaQubits/src/GateSet.qs

```diff
@@ -0,0 +1,65 @@
+/// A set of gates built upon the custom measurements
+/// provided by the hardware provider.
+///
+/// Source:
+///  [1] Surface code compilation via edge-disjoint paths
+///      https://arxiv.org/pdf/2110.11493
+
+/// Apply a CNOT gate to the given qubits.
+/// Source: [1] Figure 3.
+operation CNOT(control : Qubit, target : Qubit) : Unit {
+    // Prepare an ancilla qubit in the |+> state.
+    use ancilla = Qubit();
+    H(ancilla);
```

Comment:
In keeping with the measurement based compuation theme, this can be Mx with a conditional Z:

```suggestion
    if Mx(ancilla) == One {
        Z(ancilla);
    }
```

Reply:
I went with your suggestion, but wrapped that logic in a `PreparePlus` operation.